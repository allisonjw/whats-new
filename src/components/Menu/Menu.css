nav {
    display: flex;
    flex-direction: column;
    width: 20%;
}

button {
    height: 40px;
    width: 70%;
    color: indianred;
    border: none;
}