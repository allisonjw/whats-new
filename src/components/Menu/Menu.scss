@import './variables.scss';

.heading {
    display: flex;
    flex-direction: column;
    width: 100%;
}

h1 {
    color: white;
    font-size: 4em;
    margin-bottom: 20px;
}

nav {
    display: flex;
    flex-direction: row;
    margin-bottom: 14px;
}

.menu__btn {
    background-color: $dark_background;
    border: none;
    color: white;
    font-size: 1.5em;
    font-weight: 800;
    height: 40px;
    margin-right: 16px;
    outline: none;
    width: 41%;
}

.menu__btn:hover {
    background-color: $hover_color;
    border-radius: 18px 0 0 18px;
    cursor: pointer;
    font-weight: 900;
    transform: scale(1.1);
    transition: all .1s ease-in-out;
}