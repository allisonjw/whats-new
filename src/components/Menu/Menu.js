import React from 'react';
import './Menu.scss'
import { MdLocationOn } from "react-icons/md";
import { GiLightBulb } from "react-icons/gi";
import { MdLocalMovies} from "react-icons/md";
import { IoIosRocket } from "react-icons/io";
import { GiHeartBeats} from "react-icons/gi";

const Menu = props => {
    return (
        <div className="heading">
          <h1>What's New?</h1>  
        <nav>
            <button 
              className='menu__btn' 
              name='local' 
              onClick={e => props.changeNewsType(e)}>Local News <MdLocationOn /></button>
            <button 
              className='menu__btn' 
              name='technology' 
              onClick={e => props.changeNewsType(e)}>Technology <GiLightBulb /></button>
            <button 
              className='menu__btn' name='entertainment'
              onClick={e => props.changeNewsType(e)}>Entertainment <MdLocalMovies /></button>
            <button 
              className='menu__btn' 
              name='science'
              onClick={e => props.changeNewsType(e)}>Science <IoIosRocket /></button>
            <button 
              className='menu__btn' 
              name='health'
              onClick={e => props.changeNewsType(e)}>Health <GiHeartBeats /></button>
        </nav>
        </div>
    )
}

export default Menu;