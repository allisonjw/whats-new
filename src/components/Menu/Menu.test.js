import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {
  let mockChangeNewsType = jest.fn();
  let wrapper = shallow(<Menu changeNewsType={mockChangeNewsType} />);

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the changeNewsType prop when clicked', () => {
    wrapper.find('button').at(0).simulate('click');
    expect(mockChangeNewsType).toHaveBeenCalled();
  });

  it('should call the changeNewsType prop when clicked', () => {
    wrapper.find('button').at(1).simulate('click');
    expect(mockChangeNewsType).toHaveBeenCalled();
  });

  it('should call the changeNewsType prop when clicked', () => {
    wrapper.find('button').at(2).simulate('click');
    expect(mockChangeNewsType).toHaveBeenCalled();
  });

  it('should call the changeNewsType prop when clicked', () => {
    wrapper.find('button').at(3).simulate('click');
    expect(mockChangeNewsType).toHaveBeenCalled();
  });

  it('should call the changeNewsType prop when clicked', () => {
    wrapper.find('button').at(4).simulate('click');
    expect(mockChangeNewsType).toHaveBeenCalled();
  });
});