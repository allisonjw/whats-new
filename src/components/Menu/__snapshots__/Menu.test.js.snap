// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot 1`] = `
<div
  className="heading"
>
  <h1>
    What's
    <span>
      New?
    </span>
  </h1>
  <nav>
    <button
      className="menu__btn"
      name="local"
      onClick={[Function]}
      type="submit"
    >
      Local News
      <MdLocationOn />
    </button>
    <button
      className="menu__btn"
      name="technology"
      onClick={[Function]}
      type="submit"
    >
      Technology
      <GiLightBulb />
    </button>
    <button
      className="menu__btn"
      name="entertainment"
      onClick={[Function]}
      type="submit"
    >
      Entertainment 
      <MdLocalMovies />
    </button>
    <button
      className="menu__btn"
      name="science"
      onClick={[Function]}
      type="submit"
    >
      Science
      <IoIosRocket />
    </button>
    <button
      className="menu__btn"
      name="health"
      onClick={[Function]}
      type="submit"
    >
      Health
      <GiHeartBeats />
    </button>
  </nav>
</div>
`;
